lib/internal/fs/rimraf.js

```diff
@@ -280,17 +280,15 @@ function fixWinEPERMSync(path, options, originalErr) {
   let stats;
 
   try {
-    stats = statSync(path);
-  } catch (err) {
-    if (err.code === 'ENOENT')
-      return;
 
+    stats = statSync(path, { throwIfNoEntry: false });
+  } catch {
     throw originalErr;
   }
 
-  if (stats.isDirectory())
+  if (stats?.isDirectory())
```

Comment:
Instead of checking twice if `stats` is set, perhaps just have:

```js
if (stats) {
  if (stats.isDirectory())
    _rmdirSync(path, options, originalErr);
  else
    _unlinkSync(path, options);
}
```

or:


```js
if (!stats)
  return;
```

after the assignment above and remove these two changes.